packages/material-ui/src/Button/Button.js

```diff
@@ -258,6 +260,63 @@ export const styles = (theme) => ({
   },
 });
 
+const isSupportedColor = (color) => {
+  const supportedColors = [
+    'primary',
+    'secondary',
+    'inherit',
+  ];
+
+  return supportedColors.indexOf(color) >= 0;
+}
+
+const useColorStyles = makeStyles((theme) => ({
+  containedColor: {
+    color: ({ color }) => theme.palette[color].contrastText,
+    backgroundColor: ({ color }) => theme.palette[color].main,
+    '&:hover': {
+      backgroundColor: ({ color }) => theme.palette[color].dark,
```

Comment:
On the Figma asset, we have had a user asking about the usage of the theme with the styles.

![Screenshot 2020-07-13 at 10 43 19](https://user-images.githubusercontent.com/3165635/87365885-a8ae5980-c577-11ea-95e7-832f5e175b1a.png)

I wonder if we should use

```jsx
  backgroundColor: fade(theme.palette[color].main, theme.palette.action.hoverOpacity),
```

here and even move this style to the palette. This is related to https://github.com/mui-org/material-ui/issues/10870. We have also wondered as it could make CSS variable support easier